subprojects/core/src/test/groovy/org/gradle/api/internal/AbstractDomainObjectCollectionSpec.groovy

```diff
@@ -1431,4 +1420,339 @@ abstract class AbstractDomainObjectCollectionSpec<T> extends Specification {
         1 * action.execute(a)
         0 * action.execute(_)
     }
+
+    abstract static class AbstractConfigurationFactory<T> implements Factory<T> {
```

Comment:
I would consider pulling these classes out into a fixture class or something.  It's a lot of additional noise in an already noisy test class.